src/dma.c

```diff
@@ -47,6 +47,54 @@ static volatile int __dma_busy(void)
     return PI_regs->status & (PI_STATUS_DMA_BUSY | PI_STATUS_IO_BUSY);
 }
 
+/**
+ * @brief Check whether the specified PI address can be accessed doing I/O from CPU
+ * 
+ * The PI bus covers the full 32-bit address range. The full range is only accessible
+ * via DMA, though. A part of the range is also memory mapped to the CPU and can be
+ * accessed via #io_read and #io_write.
+ * 
+ * The ranges of PI address that can be access via CPU are:
+ * 
+ *  * 0x0500_0000 - 0x0FFF_FFFF: used by N64DD and SRAM on cartridge
+ *  * 0x1000_0000 - 0x1FBF_FFFF: cartridge ROM
+ *  * 0x1FD0_0000 - 0x1FFF_FFFF: no known PI peripherals use this
+ * 
+ * The rest of the 32-bit address range is only accessible via DMA.
+ * 
+ * Notice also that the range 0x2000_0000 - 0x7FFF_FFFF is theoretically accessible
+ * by the CPU but only via 64-bit addressing, so it requires assembly instructions
+ * (as the libdragon toolchain uses 32-bit pointers). No known PI peripherals use this
+ * range anyway.
+ * 
+ * This function checks whether the specified address falls into the range accessible
+ * via CPU or not.
+ * 
+ * @param pi_address        PI address to check
+ * @return                  True if the address is memory mapped, false if it is not
+ */
+bool io_accessible(uint32_t pi_address)
```

Comment:
What about `io_cpu_accessible`? I think it is less confusing. Otherwise people might think it is doing something else just by looking at the name.

Reply:
The rationale here is to stay similar to `io_read` / `io_write`. I can add the `cpu` work in there if you feel it's better, even though it's missing from `io_read` / `io_write`.

Reply:
looks good as is for consistency 👍